cmd/srcd/cmd/defered_test.go

```diff
@@ -1,47 +1,52 @@
 package cmd
 
 import (
-	"bytes"
-	"encoding/json"
-	"io"
-	"strings"
+	"fmt"
+	"io/ioutil"
 	"testing"
 	"time"
 
-	"github.com/sirupsen/logrus"
 	"github.com/stretchr/testify/require"
 	"github.com/stretchr/testify/suite"
+	"gopkg.in/src-d/go-log.v1"
 )
 
 type DeferedTestSuite struct {
 	suite.Suite
+
+	mockLogger *mockLogger
 }
 
 func TestDeferedTestSuite(t *testing.T) {
 	suite.Run(t, new(DeferedTestSuite))
 }
 
-func (s *DeferedTestSuite) buildFn(d *Defered, timeout time.Duration) func() {
+func (s *DeferedTestSuite) logAction(d *Defered, timeout time.Duration) {
 	s.T().Helper()
 
-	return func() {
-		logrus.Info("Start")
-		cancel := d.Print()
-		time.Sleep(timeout)
-		cancel()
-		logrus.Info("End")
-	}
+	s.mockLogger.Infof("Start")
+	cancel := d.Print()
+	time.Sleep(timeout)
+	cancel()
+	s.mockLogger.Infof("End")
 }
 
 func (s *DeferedTestSuite) buildDefered(withSpinner bool, inputFn func(stop <-chan bool) <-chan string) *Defered {
 	s.T().Helper()
 
+	l := mockLogger{}
```

Comment:
nit pick. You don't need this variable. You can do `s.mockLogger = &mockLogger{}` and `logger: s.mockLogger` below